caluma/caluma_core/models.py

```diff
@@ -10,11 +10,10 @@ def _history_user_getter(historical_instance):
 
 def _history_user_setter(historical_instance, user):
     request = getattr(HistoricalRecords.thread, "request", None)
-    user = None
+    user = "AnonymousUser"
     if request is not None:
-        user = request.user.username
-        if request.user.__class__.__name__ == "AnonymousUser":
-            user = "AnonymousUser"
+        if hasattr(request, "user"):
+            user = request.user.username
```

Comment:
```suggestion
        if hasattr(request, "user"):
            user = request.user.username or "AnonymousUser"
```

This should clean up the failing tests, as otherwise, we're now setting the user to `None` in some cases where it was `"AnonymousUser"` before